tools/test_modules/m26700.pm

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env perl
+
+##
+## Author......: See docs/credits.txt
+## License.....: MIT
+##
+
+use strict;
+use warnings;
+
+use Crypt::PBKDF2;
+use Crypt::CBC;
+use Crypt::Mode::CBC;
+
+sub module_constraints { [[0, 256], [32, 32], [-1, -1], [-1, -1], [-1, -1]] }
+
+sub module_generate_hash
+{
+  my $word = shift;
+  my $salt_str = shift // random_hex_string (32);
+  my $ct_str = shift;
+
+  my $iv_str = "";
+
+  if (defined $ct_str)
+  {
+    $iv_str = substr($ct_str, 0, 32);
+  }
+  else
+  {
+    $iv_str = random_hex_string (32);
+  }
+
+  my $salt = pack ("H*", $salt_str);
+
+  my $iv = pack ("H*", $iv_str);
+
+  my $iterations = 10000;
+
+  my $hasher = Crypt::PBKDF2->hasher_from_algorithm ('HMACSHA1');
+
+  my $pbkdf2 = Crypt::PBKDF2->new (
+    hasher       => $hasher,
+    iterations   => $iterations,
+    output_len   => 32
+  );
+
+  my $key = $pbkdf2->PBKDF2 ($salt, $word);
+
+  my $cipher = Crypt::CBC->new ({
+    key         => $key,
+    cipher      => "Crypt::Rijndael",
+    iv          => $iv,
+    literal_key => 1,
+    header      => "none",
+    keysize     => 32
+  });
+
+  my $hash = "";
+
+  if (defined $ct_str)
+  {
+    my $ct_bin = pack ("H*", $ct_str);
+    my $iv_bin = substr ($ct_bin, 0, 16);
+    my $encrypted = substr ($ct_bin, 16);
+
+    $hash = sprintf ("\$vmx\$0\$%s\$%s\$%s%s", $iterations, unpack ("H*", $salt), unpack ("H*", $iv_bin), unpack ("H*", $encrypted));
```

Comment:
Here's a fundamental error. To reproduce:

```
root@ht:~/hashcat# cat hash
$vmx$0$10000$41326347039781188312527423234988$60820f116b18406de6d2fe41d3b50caead2262d22b22924cbe3eb13a1ff4a1db8ad3fe403fb5f41f429c1c1456a426ce
root@ht:~/hashcat# cat crack 
$vmx$0$10000$41326347039781188312527423234988$60820f116b18406de6d2fe41d3b50caead2262d22b22924cbe3eb13a1ff4a1db8ad3fe403fb5f41f429c1c1456a426ce:hashcat
root@ht:~/hashcat# tools/test.pl verify 26700 hash crack x            
root@ht:~/hashcat# cat x
root@ht:~/hashcat# 
```

The x file should contain the same line as the crack file. Typically, inside the (defined) block, you decrypt, check if the decryption was successful or not. If it was successful, reuse the same data that you received as input with the idea that the script will produce the same hash line as the input line was. If it was unsuccessful, empty the data so that the resulting hash will be a different one.

Reply:
fixed, the bug was in the verify

Reply:
I don't think this is the right way.

If the password in "crack" is invalid, the output will still be the correct hash line. So there's no true verification. This part is relevant for automated processing. Please update the unit test so that with a wrong password, there will be a wrong hash (which then will lead to an empty line in the output file automatically).

Reply:
for the ST_HASH, the full buffer including the IV is 

cbb368564d8dfb99f509d4922f4693413f3816af713f0e76bc2409ff9336935d069b1c0763c06ec30bd340d87849da7db5c946f6c2bbcc2cf47fcd01d704e4711310f2e09189e2b87ada80573cafade6f83c7a27a3ceca9c5722ba7c083b358151247bb9b13fb01ad5a98ec893d8a40cfb8db06c

The truncated one is:

cbb368564d8dfb99f509d4922f4693413f3816af713f0e76bc2409ff9336935d

Reply:
> I don't think this is the right way.
> 
> If the password in "crack" is invalid, the output will still be the correct hash line. So there's no true verification. This part is relevant for automated processing. Please update the unit test so that with a wrong password, there will be a wrong hash (which then will lead to an empty line in the output file automatically).

done